common/membership/hashring.go

```diff
@@ -264,6 +265,16 @@ func (r *ring) refresh() (refreshed bool, err error) {
 	return true, nil
 }
 
+func (r *ring) refreshAndNotifySubscribers(event *ChangedEvent) {
```

Comment:
I can see history's shard/controller listens these events and proactively starting the engines ([code](https://github.com/uber/cadence/blob/8871abcd798b8767b7ea0ed299be98b1be5c46bb/service/history/shard/controller.go#L377)). 
There's not a corresponding logic on matching side. Tasklist managers are created when first add/poll request arrive. This might be OK for most cases but we can also consider adding a similar proactive tasklist manager starter based on ring state

Reply:
We have this in matching since recently: [membership.go](https://github.com/uber/cadence/blob/04add2d18e25943498cfe97c93dde55b114e4955/service/matching/handler/membership.go#L66) is responsible for that.

Reply:
Exactly. on-demand creation relies on a addtask or polltask request to ensure a tasklistmanager is alive. It should be fine because if there's no poller then no need to create a tasklistmanager.

Reply:
The thing is - if we don't get notification we won't stop existing task-list-manager(s) we do not own anymore. This includes not disconnecting pollers who are waiting for match in those task-list-managers which results in spike of schedule-to-start every time cadence-matching instances restart/redeployed.

Reply:
Yes this will help the proactive tasklist shutdown (introduced in #6198) because it relies on `Subscribe` call.
Prpactive tasklist creation still doesn't happen - it will stay as on-demand which is probably fine. We don't have a list of "active" tasklists somewhere to find and start them anyway.